crates/torii/grpc/src/server/mod.rs

```diff
@@ -227,6 +215,95 @@ impl DojoWorld {
         row_events.iter().map(map_row_to_event).collect()
     }
 
+    async fn fetch_entities(
+        &self,
+        table: &str,
+        entity_relation_column: &str,
+        entities: Vec<(String, String)>,
+    ) -> Result<Vec<proto::types::Entity>, Error> {
+        // Group entities by their model combinations
+        let mut model_groups: HashMap<String, Vec<String>> = HashMap::new();
+        for (entity_id, models_str) in entities {
+            model_groups.entry(models_str).or_default().push(entity_id);
+        }
+
+        let mut all_entities = Vec::new();
+
+        // Start a transaction
+        let mut tx = self.pool.begin().await?;
+
+        // Create a temporary table to store entity IDs
```

Comment:
Changing this by mentioned the parameters limit and the reason of using this temporary table.